modules/terminate-workers/lambda/lambda_function.py

```diff
@@ -0,0 +1,120 @@
+"""
+AWS Lambda function to terminate orphaned GitLab runners.
+
+This checks for running GitLab runner worker instances and terminates them,
+intended to be triggered by an ASG life cycle hook at instance termination.
+
+https://github.com/npalm/terraform-aws-gitlab-runner/issues/317 has some
+discussion about this scenario.
+
+This is rudimentary and doesn't check if a build runner has a current job.
+"""
+import boto3
+import json
+
+def ec2_list(client, **args):
+
+    print(json.dumps({
+        "Level": "info",
+        "Message": f"Searching for children of GitLab runner instance {args['parent']}"
+    }))
+
+    ec2_instances = client.describe_instances(Filters=[
+            {
+                "Name": "instance-state-name",
+                "Values": ['running'],
```

Comment:
question: Is this sufficient? What if a new instance is being created? We should kill it too.

Reply:
I added "pending" to this list in a290efa4934c4a43c3f14143be6f21d5ba1ef689